lxd/devices.go

```diff
@@ -62,10 +62,31 @@ type deviceTaskCPU struct {
 
 type deviceTaskCPUs []deviceTaskCPU
 
+var numaNodeToCPU map[int64][]int64
+
 func (c deviceTaskCPUs) Len() int           { return len(c) }
 func (c deviceTaskCPUs) Less(i, j int) bool { return *c[i].count < *c[j].count }
 func (c deviceTaskCPUs) Swap(i, j int)      { c[i], c[j] = c[j], c[i] }
 
+func init() {
```

Comment:
I think these type of global initialisation functions are only really acceptable when they cant fail  (given the only thing you can do is log the error as you have done here).

What was the reason you went down the caching route here? (for example there are a few places where resources.GetCPU() is called already in LXD without caching).

Also, if we do decide to go with the caching approach, how will we deal with cache invalidation/refresh when CPU counts change (for example CPU hotplugging)?

Reply:
Originally, I thought about the caching approach because I must admit I didn't think about the CPU hotplugging case :) so now I'll get back to using `resources.GetCPU()` at the start of the `deviceTaskBalance` function